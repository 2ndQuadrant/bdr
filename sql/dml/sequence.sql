CREATE SEQUENCE bdr_test_seq USING bdr;

SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), 0);

DO $$
BEGIN
	LOOP
		IF (SELECT amdata IS NOT NULL FROM bdr_test_seq) THEN
			EXIT;
		END IF;
		PERFORM pg_sleep(0.1);
	END LOOP;
END;$$;

CREATE TEMP TABLE bdr_test_seq_val AS SELECT nextval('bdr_test_seq') as last_val;

SELECT bdr.bdr_internal_sequence_reset_cache('bdr_test_seq'::regclass);
SELECT amdata IS NULL FROM bdr_test_seq;

DO $$
BEGIN
	LOOP
		IF (SELECT amdata IS NOT NULL FROM bdr_test_seq) THEN
			EXIT;
		END IF;
		PERFORM pg_sleep(0.1);
	END LOOP;
END;$$;

-- 10000 is maximum cache size in amdata
SELECT nextval('bdr_test_seq') >= last_val + 10000 FROM bdr_test_seq_val;
