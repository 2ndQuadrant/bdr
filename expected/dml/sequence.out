CREATE SEQUENCE bdr_test_seq USING bdr;
SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), 0);
 pg_xlog_wait_remote_apply 
---------------------------
 
(1 row)

DO $$
BEGIN
	LOOP
		IF (SELECT amdata IS NOT NULL FROM bdr_test_seq) THEN
			EXIT;
		END IF;
		PERFORM pg_sleep(0.1);
	END LOOP;
END;$$;
CREATE TEMP TABLE bdr_test_seq_val AS SELECT nextval('bdr_test_seq') as last_val;
SELECT bdr.bdr_internal_sequence_reset_cache('bdr_test_seq'::regclass);
 bdr_internal_sequence_reset_cache 
-----------------------------------
 
(1 row)

SELECT amdata IS NULL FROM bdr_test_seq;
 ?column? 
----------
 t
(1 row)

DO $$
BEGIN
	LOOP
		IF (SELECT amdata IS NOT NULL FROM bdr_test_seq) THEN
			EXIT;
		END IF;
		PERFORM pg_sleep(0.1);
	END LOOP;
END;$$;
-- 10000 is maximum cache size in amdata
SELECT nextval('bdr_test_seq') >= last_val + 10000 FROM bdr_test_seq_val;
 ?column? 
----------
 t
(1 row)

